File: GoNexus-Web/src/api/moment.ts
import request from '@/utils/request'

export interface CreatePostRequest {
  content: string
  media?: string[]
  visibility: number // 0-公开, 1-好友, 2-私密
  location?: string
  mood?: string
}

export interface Comment {
  id: number
  post_id: number
  user_id: number
  user_nickname: string
  user_avatar: string
  content: string
  parent_id?: number
  created_at: string
}

export interface Post {
  id: number
  user_id: number
  user_nickname: string
  user_avatar: string
  content: string
  media: string[]
  visibility: number
  location: string
  mood: string
  like_count: number
  comment_count: number
  is_liked: boolean
  created_at: string
  comments?: Comment[]
}

// 发布动态
export function createPost(data: CreatePostRequest) {
  return request({
    url: '/moment/post',
    method: 'post',
    data
  })
}

// 获取动态列表
export function getMoments(params: { page: number; page_size: number; type: string; user_id?: number }) {
  return request({
    url: '/moment/list',
    method: 'get',
    params
  })
}

// 发表评论
export function createComment(data: { post_id: number; content: string; parent_id?: number }) {
  return request({
    url: '/moment/comment',
    method: 'post',
    data
  })
}

// 获取评论
export function getComments(post_id: number) {
  return request({
    url: '/moment/comments',
    method: 'get',
    params: { post_id }
  })
}

// 点赞/取消点赞
export function toggleLike(post_id: number) {
  return request({
    url: '/moment/like',
    method: 'post',
    data: { post_id }
  })
}

File: GoNexus/internal/model/dto/moment.go
package dto

import "time"

type CreatePostRequest struct {
	Content    string   `json:"content" binding:"required"`
	Media      []string `json:"media"`
	Visibility int      `json:"visibility"` // 0-公开, 1-好友, 2-私密
	Location   string   `json:"location"`
	Mood       string   `json:"mood"`
}

type CreateCommentRequest struct {
	PostID  uint   `json:"post_id" binding:"required"`
	Content string `json:"content" binding:"required"`
	ParentID *uint `json:"parent_id"`
}

type LikePostRequest struct {
	PostID uint `json:"post_id" binding:"required"`
}

type GetMomentsRequest struct {
	Page     int    `form:"page,default=1"`
	PageSize int    `form:"page_size,default=10"`
	Type     string `form:"type"` // "all" (广场) or "friend" (朋友圈) or "user" (个人)
	UserID   uint   `form:"user_id"` // 查看特定用户的动态
}

type PostResponse struct {
	ID           uint      `json:"id"`
	UserID       uint      `json:"user_id"`
	UserNickname string    `json:"user_nickname"`
	UserAvatar   string    `json:"user_avatar"`
	Content      string    `json:"content"`
	Media        []string  `json:"media"`
	Visibility   int       `json:"visibility"`
	Location     string    `json:"location"`
	Mood         string    `json:"mood"`
	LikeCount    int       `json:"like_count"`
	CommentCount int       `json:"comment_count"`
	IsLiked      bool      `json:"is_liked"` // 当前用户是否已点赞
	CreatedAt    time.Time `json:"created_at"`
	Comments     []CommentResponse `json:"comments,omitempty"` // 仅在详情页或预加载时返回
}

type CommentResponse struct {
	ID           uint      `json:"id"`
	PostID       uint      `json:"post_id"`
	UserID       uint      `json:"user_id"`
	UserNickname string    `json:"user_nickname"`
	UserAvatar   string    `json:"user_avatar"`
	Content      string    `json:"content"`
	ParentID     *uint     `json:"parent_id"`
	CreatedAt    time.Time `json:"created_at"`
}
